cmd/healthcheck/framework/healthcheck.go

```diff
@@ -0,0 +1,445 @@
+	}
+
+	if instance == nil {
+		return h.setError("error creating instance - instance is null", "")
+	}
+
+	glog.V(4).Info("Waiting for ServiceInstance to be ready")
+	err = util.WaitForInstanceCondition(h.serviceCatalogClientSet.ServicecatalogV1beta1(),
+		h.namespace.Name,
+		h.instanceName,
+		v1beta1.ServiceInstanceCondition{
+			Type:   v1beta1.ServiceInstanceConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		return h.setError("instance not ready: %v", err.Error())
+	}
+	ReportOperationCompleted("create_instance", operationStartTime)
+
+	glog.V(4).Info("Verifing references are resolved")
+	sc, err := h.serviceCatalogClientSet.ServicecatalogV1beta1().ServiceInstances(h.namespace.Name).Get(h.instanceName, metav1.GetOptions{})
+	if err != nil {
+		return h.setError("error getting instance: %v", err.Error())
+	}
+
+	if sc.Spec.ClusterServiceClassRef == nil {
+		return h.setError("ClusterServiceClassRef should not be null", "")
+	}
+	if sc.Spec.ClusterServicePlanRef == nil {
+		return h.setError("ClusterServicePlanRef should not be null", "")
+	}
+
+	if strings.Compare(sc.Spec.ClusterServiceClassRef.Name, h.serviceclassID) != 0 {
+		return h.setError("ClusterServiceClassRef.Name should not be null", "")
+	}
+	if strings.Compare(sc.Spec.ClusterServicePlanRef.Name, h.serviceplanID) != 0 {
+		return h.setError("ClusterServicePlanRef.Name should not be null", "")
+	}
+	return nil
+}
+
+// createBinding creates a binding and verifies the binding and secret are
+// correct
+func (h *HealthCheck) createBinding() error {
+	if h.frameworkError != nil {
+		return h.frameworkError
+	}
+	glog.V(4).Info("Creating a ServiceBinding")
+	binding := &v1beta1.ServiceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      h.bindingName,
+			Namespace: h.namespace.Name,
+		},
+		Spec: v1beta1.ServiceBindingSpec{
+			ServiceInstanceRef: v1beta1.LocalObjectReference{
+				Name: h.instanceName,
+			},
+			SecretName: "my-secret",
+		},
+	}
+	operationStartTime := time.Now()
+	binding, err := h.serviceCatalogClientSet.ServicecatalogV1beta1().ServiceBindings(h.namespace.Name).Create(binding)
+	if err != nil {
+		return h.setError("Error creating binding: %v", err.Error())
+	}
+	if binding == nil {
+		return h.setError("Binding should not be null", "")
+	}
+
+	glog.V(4).Info("Waiting for ServiceBinding to be ready")
+	_, err = util.WaitForBindingCondition(h.serviceCatalogClientSet.ServicecatalogV1beta1(),
+		h.namespace.Name,
+		h.bindingName,
+		v1beta1.ServiceBindingCondition{
+			Type:   v1beta1.ServiceBindingConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		return h.setError("binding not ready: %v", err.Error())
+	}
+	ReportOperationCompleted("binding_ready", operationStartTime)
+
+	glog.V(4).Info("Validating that a secret was created after binding")
+	_, err = h.kubeClientSet.CoreV1().Secrets(h.namespace.Name).Get("my-secret", metav1.GetOptions{})
```

Comment:
It would be useful as a follow-on to check that the secret has the expected keys.